Use nightly changelog theme

wilzbach opened the pull request:
It seems like we are stuck with the `VERSION` macro.
The best solution I found is to overwrite temporarily it with `NIGHTLY_VERSION`, however this depends on https://github.com/dlang/tools/pull/221

To make it "stable" (aka without the nightly information), the easiest way is to set `BUILD_NIGHTLY=0`

CyberShadow commented:
Awesome!

CyberShadow commented on posix.mak:
```diff
@@ -42,6 +42,9 @@ STABLE_DMD_CONF=$(STABLE_DMD).conf
 	--compiler=$(STABLE_DMD) -conf=$(STABLE_DMD_CONF)
 
+# whether to use a special theme for nightly changelog builds
+BUILD_NIGHTLY=1
```
Why is this option needed?

wilzbach replied:
The idea was that Martin can call the Makefile with `BUILD_NIGHTLY=0` and then it will generate the to be released page without the nightly texts. In other words:
- current date instead of "to be released"
- `VERSION` instead of `NIGHTLY_VERSION`
- no `(upcoming)` in the title
- ...

wilzbach replied:
See also https://github.com/dlang/tools/pull/222 for an example on more "customization" based on `BUILD_NIGHTLY`

wilzbach commented:
>> OK, LGTM and leaving this for @MartinNowak to review as it affects the release process.
> Just name the changlog _pre.d will build a pre-release changlog.

-> renamed to `changelog/${NEXT_VERSION}_pre.dd` for the default generation (nightly).

There's also a `changelog/${NEXT_VERSION}.dd` target, which can be used by to generate the "release version" of the changelog (without "to be released", nightlies download link, ...).

As we now have two different files I added the `_pre` files to the `.gitignore`, s.t. a default build of `dlang.org` doesn't generate any "new files".

> Always using stable seems incorrect when the purpose of this target is trying to build the changelog for dmd/druntime/phobos PRs.

I have changed it, s.t. for the `_pre` changelog files `upstream/master` is used instead.
Due to the new `DIFFABLE` Makefile variable this doesn't matter on DAutoTest anyways.

CyberShadow commented:
@MartinNowak If you're happy with these could you please merge them? Thanks!

MartinNowak commented:
THX

MartinNowak commented on posix.mak:
```diff
@@ -588,12 +589,16 @@ test:
+	$(STABLE_RDMD) $(TOOLS_DIR)/changed.d $(CHANGELOG_VERSION_MASTER) -o $@ \
+	--version "${NEXT_VERSION} (upcoming)" --date "To be released" --nightly
+
 changelog/${NEXT_VERSION}.dd: ${STABLE_DMD} ../tools ../installer
```
Do we still need that rule for anything?
